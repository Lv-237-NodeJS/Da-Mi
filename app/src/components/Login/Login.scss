.login-form {
  height: 200px;
  margin: 10px 0 50px;
}
