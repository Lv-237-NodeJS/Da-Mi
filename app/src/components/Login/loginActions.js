import { browserHistory } from 'react-router';
import { API, request } from 'src/helper';

const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';
const LOGOUT_USER = 'LOGOUT_USER';
const CHECK_TOKEN = 'CHECK_TOKEN';

export const loginUserSuccess = (token, userId, profileId) => {
  sessionStorage.setItem('token', token);
  sessionStorage.setItem('userId', userId);
  sessionStorage.setItem('profileId', profileId);
  browserHistory.push('/events');
  return {
    type: LOGIN_USER_SUCCESS,
    userId: userId,
    profileId: profileId
  };
};

export const loginUserFailure = message => ({
  type: LOGIN_USER_FAILURE,
  message
});

export const checkToken = () => ({
  type: CHECK_TOKEN,
  isAuth: !!sessionStorage.getItem('token')
});

export const logout = () => {
  sessionStorage.clear();
  browserHistory.push('/');
  return {
    type: LOGOUT_USER
  };
};

export const loginUser = (email, password) => {
  const user = {email, password};
  let token;
  let userId;
  let profileId;
  return dispatch => request()
    .post(`${API.URL}/api/auth/login`)
    .send(user)
    .end((err, res) => {
      (err || !res.ok) &&
      dispatch(loginUserFailure(JSON.parse(res.text).message)) ||
      ({token, user_id: userId, profile_id: profileId} = JSON.parse(res.text)) &&
      dispatch(loginUserSuccess(token, userId, profileId));
    });
};
