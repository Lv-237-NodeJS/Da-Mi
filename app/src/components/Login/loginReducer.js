const loginReducer = {
  LOGIN_USER_FAILURE: (state, action) => {
    return {
      ...state,
      illegalInput: true,
      message: action.message
    };
  },
  LOGIN_USER_SUCCESS: (state, action) => {
    return {
      ...state,
      illegalInput: false,
      isAuth: true,
      userId: action.userId
    };
  },
  LOGOUT_USER: (state, action) => {
    return {
      ...state,
      isAuth: false
    };
  },
  CHECK_TOKEN: (state, action) => {
    return {
      ...state,
      isAuth: action.isAuth
    };
  }
};

export default loginReducer;
