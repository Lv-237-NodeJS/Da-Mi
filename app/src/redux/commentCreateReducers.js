import  { API, request }  from 'src/helper';

const COMMENT_CREATED = 'COMMENT_CREATED';
const CREATE_COMMENT_FAILURE = 'CREATE_COMMENT_FAILURE';

export const createComment = (eventId, giftId, commentData) => dispatch =>
  request()
    .post(`${API.URL}/api/event/${eventId}/gift/${giftId}/comments`)
    .send(commentData)
    .end((err, res) => err &&
        dispatch({
          type: CREATE_COMMENT_FAILURE,
          payload: err,
        }) ||
        dispatch({
          type: COMMENT_CREATED,
          payload: res.body,
        }));

export const createCommentReducer = (state = { comment: { } }, action) => {
  switch (action.type) {
    case COMMENT_CREATED: {
      return {
        ...state,
        comment: action.payload,
      };
    }

    case CREATE_COMMENT_FAILURE: {
      return {
        ...state,
        error: action.payload,
      };
    }

    default: return state;
  }
};

export default createCommentReducer;
