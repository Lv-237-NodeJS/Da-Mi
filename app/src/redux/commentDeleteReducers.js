const COMMENT_DELETED = 'COMMENT_DELETED';
const DELETE_COMMENT_FAILURE = 'DELETE_COMMENT_FAILURE';
import  { API }  from './../helper/constants';
import request from './../helper/request';

export const deleteComment = (eventId, giftId, comment_id) => {  
  return dispatch => {
    request()
      .delete(API.URL + `/api/event/${eventId}/gift/${giftId}/comment/${comment_id}`)      
      .end((err, res) => {
        if (err) {
          dispatch({
            type: DELETE_COMMENT_FAILURE,
            payload: err,
          });
        } else {
          dispatch({
            type: COMMENT_DELETED,
            payload: res.body,
          });
        }
      });
  };
};

export const deleteCommentReducer = (state = {commentDeleted: {}}, action) => {
  switch (action.type) {    
    case COMMENT_DELETED: {      
      return {
        ...state,
        commentDeleted: action.payload,
      };
    }
    case DELETE_COMMENT_FAILURE: {
      return {
        ...state,
        error: action.payload,
      };
    }
    default: return state;
  }
};

export default deleteCommentReducer;
