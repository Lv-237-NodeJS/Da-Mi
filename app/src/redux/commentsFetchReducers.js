import  { API, request }  from 'src/helper';

const RETRIEVE_COMMENTS = 'RETRIEVE_COMMENTS';
const RETRIEVE_COMMENTS_SUCCESS = 'RETRIEVE_COMMENTS_SUCCESS';
const RETRIEVE_COMMENTS_FAILURE = 'RETRIEVE_COMMENTS_FAILURE';

export const retrieveComments = (eventId, giftId) => dispatch =>
  request()
    .get(`${API.URL}/api/event/${eventId}/gift/${giftId}/comments`)
    .end((err, res) => err &&
        dispatch({
          type: RETRIEVE_COMMENTS_FAILURE,
        }) ||
        dispatch({
          type: RETRIEVE_COMMENTS_SUCCESS,
          payload: res.body.data,
        }));

export const CommentsReducers = (state = { comments: [] }, action) => {
  switch (action.type) {
    case RETRIEVE_COMMENTS: {
      return {
        ...state,
      };
    }

    case RETRIEVE_COMMENTS_SUCCESS: {
      return {
        ...state,
        comments: action.payload,
      };
    }

    case RETRIEVE_COMMENTS_FAILURE: {
      return {
        ...state,
      };
    }

    default: return state;
  }
};

export default CommentsReducers;
