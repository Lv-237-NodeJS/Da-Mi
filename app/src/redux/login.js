import request from './../helper/request';
import { API } from './../helper/constants';
import { browserHistory } from 'react-router';

const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';
const LOGOUT_USER = 'LOGOUT_USER';
const LOGIN_USER_REQUEST = 'LOGIN_USER_REQUEST';
const CHECK_TOKEN = 'CHECK_TOKEN';

export default function loginReduser(state = {}, action) {
  switch (action.type) {
    case LOGIN_USER_FAILURE:
      return {
        ...state,
        illegalInput: true
      };
    case LOGIN_USER_SUCCESS:
      return {
        ...state,
        illegalInput: false,
        isAuth: true,
        userId: action.userId
      };
    case LOGOUT_USER:
      return {
        ...state,
        isAuth: false
      };
    case CHECK_TOKEN:
      return {
        ...state,
        isAuth: action.isAuth
      };
    default:
      return state;
  }
}

export function loginUserSuccess(token, userId) {
  sessionStorage.setItem('token', token);
  sessionStorage.setItem('userId', userId);
  browserHistory.push('/events');
  return {
    type: LOGIN_USER_SUCCESS,
    userId: userId
  };
}

export function loginUserFailure() {
  return {
    type: LOGIN_USER_FAILURE
  };
}

export function loginUserRequest() {
  return {
    type: LOGIN_USER_REQUEST
  };
}

export function checkToken() {
  return {
    type: CHECK_TOKEN,
    isAuth: !!sessionStorage.getItem('token')
  };
}

export function logout() {
  sessionStorage.clear();
  browserHistory.push('/');
  return {
    type: LOGOUT_USER
  };
}

export function loginUser(email, password) {
  const user = {
    email: email,
    password: password
  };
  return dispatch => {
    dispatch(loginUserRequest());
    request()
      .post(API.HOST + API.PORT + '/api/auth/login')
      .send(user)
      .end((err, res) => {
        if (err || !res.ok) {
          dispatch(loginUserFailure());
        } else {
          const token = JSON.parse(res.text).token;
          const userId = JSON.parse(res.text).user_id;
          dispatch(loginUserSuccess(token, userId));
        }
      });
  };
}
