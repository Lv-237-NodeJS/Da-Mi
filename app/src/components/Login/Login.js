import React from 'react';
import { Form, FormGroup, ControlLabel, FormControl, Col, Button }
  from 'react-bootstrap';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as loginActions from 'src/redux/login';
import './Login.scss';

const InputGroup = ({id, label, ...props}) => (
  <FormGroup controlId={id}>
    <Col componentClass={ControlLabel} sm={12} md={3}>
      {label}
    </Col>
    <Col sm={12} md={9}>
      <FormControl {...props} required/>
    </Col>
  </FormGroup>
);


export class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = stateName => e => {
    this.setState({[stateName]: e.target.value});
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.actions.loginUser(this.state.email, this.state.password);
  };

  render() {

    return (
      <div>
        <Form className='login-form' horizontal onSubmit={this.handleSubmit} data-state=
          {this.state}>
          {['email', 'password'].map(param =>
            <InputGroup
              key={param}
              id={param}
              label={param.toUpperCase()}
              type={param}
              placeholder={param.toUpperCase()}
              value={this.state.param}
              onChange={this.handleChange(param)} />
          )}
          <FormGroup>
            <Col smOffset={0} sm={12} mdOffset={3} md={9}>
              <Button type='submit' bsStyle='primary' bsSize='large'>
                Login
              </Button>
              {this.props.badInputs &&
                <p className='text-danger'>{this.props.message}</p>
              }
            </Col>
          </FormGroup>
        </Form>
        <p className='text-center'>OR</p>
        <Link to='/signup'>
          <Button bsSize='large' block>SIGN UP</Button>
        </Link>
      </div>
    );
  }
}

const mapStatetoProps = state => ({
  badInputs: state.login.illegalInput,
  message: state.login.message
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(loginActions, dispatch)
});

export default connect(mapStatetoProps, mapDispatchToProps)(Login);
