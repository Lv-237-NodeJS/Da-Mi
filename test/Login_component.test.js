import React from 'react';
import renderer from 'react-test-renderer';
import { Login } from './../app/src/components/Login/Login';
import { loginUser } from './../app/src/redux/login';

describe('React component <Login />', () => {
  it('should render a Login with typed data' +
    ', especially checked out a correct work onChange and onSubmit functions', () => {
    const user = {
      actions: {
        loginUser
      }
    };
    const component = renderer.create(
      <Login {...user} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    const form = tree.children.filter(child => child.type === 'form')[0];
    const email = form.children[0].children[1].children.filter(child => child.props.type ===
     'email')[0];
    email.props.onChange({
      target: {
        value: 'ivan.yarymovych@gmail.com'
      }
    });
    const password = form.children[1].children[1].children.filter(child => child.props.type ===
     'password')[0];
    password.props.onChange({
      target: {
        value: 'P!assword!1'
      }
    });
    form.props.onSubmit({
      preventDefault: () => {},
    });
    tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
