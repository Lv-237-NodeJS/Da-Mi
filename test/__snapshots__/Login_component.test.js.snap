// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React component <Login /> should render a Login with typed data, especially checked out a correct work onChange and onSubmit functions 1`] = `
<div>
  <form
    className="login-form form-horizontal"
    data-state={
      Object {
        "email": "",
        "password": "",
      }
    }
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="col-md-3 col-sm-12 control-label"
        htmlFor="email"
      >
        EMAIL
      </label>
      <div
        className="col-md-9 col-sm-12"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="EMAIL"
          required={true}
          type="email"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="col-md-3 col-sm-12 control-label"
        htmlFor="password"
      >
        PASSWORD
      </label>
      <div
        className="col-md-9 col-sm-12"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="PASSWORD"
          required={true}
          type="password"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-9 col-md-offset-3 col-sm-12 col-sm-offset-0"
      >
        <button
          className="btn btn-lg btn-primary"
          disabled={false}
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
  </form>
  <p
    className="text-center"
  >
    OR
  </p>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-default btn-block"
      disabled={false}
      type="button"
    >
      SIGN UP
    </button>
  </a>
</div>
`;

exports[`React component <Login /> should render a Login with typed data, especially checked out a correct work onChange and onSubmit functions 2`] = `
<div>
  <form
    className="login-form form-horizontal"
    data-state={
      Object {
        "email": "ivan.yarymovych@gmail.com",
        "password": "P!assword!1",
      }
    }
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="col-md-3 col-sm-12 control-label"
        htmlFor="email"
      >
        EMAIL
      </label>
      <div
        className="col-md-9 col-sm-12"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="EMAIL"
          required={true}
          type="email"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="col-md-3 col-sm-12 control-label"
        htmlFor="password"
      >
        PASSWORD
      </label>
      <div
        className="col-md-9 col-sm-12"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="PASSWORD"
          required={true}
          type="password"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-9 col-md-offset-3 col-sm-12 col-sm-offset-0"
      >
        <button
          className="btn btn-lg btn-primary"
          disabled={false}
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
  </form>
  <p
    className="text-center"
  >
    OR
  </p>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-default btn-block"
      disabled={false}
      type="button"
    >
      SIGN UP
    </button>
  </a>
</div>
`;
